client/js/components/user/DpMasterToebList.vue

```diff
@@ -593,6 +532,67 @@ export default {
       } else if (this.editModeElementId === 0) {
         dplan.notify.error(Translator.trans('error.api.generic'))
       }
+    },
+
+    updateOrga (ident, updatedField, status) {
+      this.updatedItems = this.updatedItems.map(item => {
+        let newItem = item
+        if (item.ident === ident) {
+          newItem = { ...newItem, ...updatedField }
+        }
+        return newItem
+      })
+
+      this.rowItems = this.rowItems.map(item => {
+        let newItem = item
+        if (item.ident === ident) {
+          newItem = { ...newItem, ...updatedField }
+        }
+        return newItem
+      })
+
+      if (status === 'confirm') {
+        const fieldName = this.fields.filter(el => el.field === Object.keys(updatedField)[0])[0].value
+        dplan.notify.notify(status, Translator.trans('confirm.field.changes.saved', { fieldName: fieldName }))
+      } else {
+        dplan.notify.notify(status, Translator.trans('error.api.generic'))
+      }
+    },
+
+    updateFields () {
+      let sortedList = this.currentItems
+
+      // Sort by selected col
+      sortedList.sort((a, b) => {
+        /*
+         * In JS is `null > 'string' === null < 'string'`
+         * so we have to do such weired stuff
+         */
+        if (a[this.sortOrder.key] === null || typeof a[this.sortOrder.key] === 'undefined') {
+          return this.sortOrder.direction * 1
+        }
+        if (b[this.sortOrder.key] === null || typeof b[this.sortOrder.key] === 'undefined') {
+          return this.sortOrder.direction * -1
+        }
+
+        return String(a[this.sortOrder.key])
+          .localeCompare(String(b[this.sortOrder.key]), 'de', { sensitivity: 'base' }) * this.sortOrder.direction
+      })
+
+      // Apply filters and search term
+      sortedList = sortedList.filter(item => this.searchedItems[item.ident] === 'in' && this.filteredItems[item.ident] === 'in')
+
+      this.rowItems = sortedList
+      this.$nextTick(() => {
+        if (this.isEditable) this.addCellIdsAndFields(this.dataTableElement.getElementsByTagName('tbody')[0], this.onPageItems)
+      })
+    },
```

Comment:
I would reduce the cognitive load here a little bit by extracting some parts of this function into own functions. This separates the different logical steps in this operation and improves the readability. Also we can use the `toSorted` array method which also returns a new array and that way we do not need to re-declare the `sortedList` variable.

```suggestion
 sortCurrentItems (items) {
      items.toSorted((a, b) => {
        if (a[this.sortOrder.key] === null || typeof a[this.sortOrder.key] === 'undefined') {
          return this.sortOrder.direction * 1
        }
        
        if (b[this.sortOrder.key] === null || typeof b[this.sortOrder.key] === 'undefined') {
          return this.sortOrder.direction * -1
        }

        return String(a[this.sortOrder.key])
          .localeCompare(String(b[this.sortOrder.key]), 'de', { sensitivity: 'base' }) * this.sortOrder.direction
      })   
    },
    
    updateFields () {
      const sortedItems = this.sortCurrentItems(this.currentItems)
      const filteredSortedItems = sortedItems.filter(item => this.searchedItems[item.ident] === 'in' && this.filteredItems[item.ident] === 'in')

      this.rowItems = filteredSortedItems
      this.$nextTick(() => {
        if (this.isEditable) this.addCellIdsAndFields(this.dataTableElement.getElementsByTagName('tbody')[0], this.onPageItems)
      })
    },
```

Reply:
Thanks for the suggestion. Done in [5a95554](https://github.com/demos-europe/demosplan-core/pull/2621/commits/5a95554434ea9f4b918f1cd4be78f63a9c2312bc). 
I changed some things:
- `sortItems` instead of `sortCurrentItems`: The function just sorts some items it is given. In our case they are the current items, but it could be any items.
- We don't need the `toSorted` function in the `sortItems` function, because we do nothing with the param items. The `this.currentItems` are not altered. But thanks for the reminder that this function exists now :)
- Removed the `filteredSortedItems`, because it was an unnecessary step

Reply:
thank you for your efforts here, this looks really clean now :heart: